pkg/ebpf/c/tracee.bpf.c

```diff
@@ -792,6 +796,24 @@ typedef struct file_info {
     u64 ctime;
 } file_info_t;
 
+typedef struct buf_data {
+    void *ptr;
+    size_t len;
+} buf_data_t;
+
+typedef struct vec_data {
+    struct iovec *ptr;
+    unsigned long vlen;
+} vec_data_t;
+
+typedef struct io_data {
+    union {
+        buf_data_t buf;
+        vec_data_t vec;
+    };
+    bool is_buf;
+} io_data_t;
+
 typedef struct proc_info {
```

Comment:
What about using:

```C
typedef struct io_data {
    union {
        struct {
            void *ptr;
            size_t len;
        } buff;
        struct {
            struct iovec *ptr;
            unsigned long vlen;
        };
    };
    bool is_buf;
} io_data_t;
```

so we have less types to manage ?

Reply:
And if you think this straight, size_t is an unsigned long, and both first members are a pointer.. so you could replace all this for:

```C
typedef struct io_data {
    void *ptr;
    bool is_buf;
    unsigned long len;
} io_data_t;
```

and remove the if/elses from `do_file_io_operation`:

```C
    if (is_buf)
        save_to_submit_buf(&data, &io_data.buf.len, sizeof(size_t), 3);
    else
        save_to_submit_buf(&data, &io_data.vec.vlen, sizeof(unsigned long), 3);
    save_to_submit_buf(&data, &start_pos, sizeof(off_t), 4);
```

no ?

Reply:
I really like this change.
Very elegant.